### PR TITLE
support multiple go versions

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
@smarterclayton ptal.

File: hack/listtests.go [hunk 1]
```diff
@@ -70,7 +70,8 @@ func main() {
 			continue
 		}
 		// ignore non code segments
-		if match[2] != "T" {
+		//fmt.Println(line)
```
Comment (smarterclayton) [inline hack/listtests.go]: Keep the println inside - debugging for someone later

**Comment by bparees** [comment]
not sure what you mean...you want to leave the println live?

**Comment by smarterclayton** [comment]
The diff is showing it outside the if, vs in

**Comment by bparees** [comment]
yeah but it was also commented out.  I hadn't even meant to leave it in the commit.  the debug value was to print the lines before they got filtered away.  

anyway i've put it back in commented out.  merge?

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by ncdc** [comment]
Nice :-)

I also have a patch for starting etcd once we move to 2.0 (which I needed in addition to this when doing some recent integration test runs locally).

**Comment by bparees** [comment]
@mfojtik @soltysh looks like the race condition is still happening: https://ci.openshift.redhat.com/jenkins/job/merge_pull_requests_openshift3/1078/consoleFull  (search for FAIL: TestSimpleImageChangeBuildTrigger)

[merge]

**Comment by bparees** [comment]
@smarterclayton  btw it kinda bothers me that if golang changes their internals again, it's possible we'll stop running all the integration tests with no indication that has happened.  maybe we need to add some logic to check if zero tests were run and if so, fail the run.

**Comment by smarterclayton** [comment]
Yes, good idea.  Also, I need to print a separator for each test so you can see what's happening (in the script).

**Comment by smarterclayton** [comment]
Someone definitely needs to get to the bottom of this.  Either it's in the controller (very bad) or its in the test (which seems unlikely given that people have looked at it).

**Pull request merged**